pyqt-apps/siriushla/li_ap_mps/main.py

```diff
@@ -0,0 +1,553 @@
+        ''' Display the temperature label widget '''
+        device_name = self.getDeviceName(pv_name)
+        widget = PyDMLabel(
+            parent=self,
+            init_channel= device_name + pv_name + temp_type
+        )
+        widget.showUnits = True
+        return widget
+
+    def clearLayout(self, lay):
+        '''Erase the layout's content'''
+        while lay.count():
+            child = lay.takeAt(0)
+            if child.widget():
+                child.widget().deleteLater()
+        return lay
+
+    def controlHiddenBox(self, pv_name, cb_glay, config):
+        '''Display the status of the PV - Hidden control Box'''
+
+        if cb_glay == '':
+            cb_glay = QGridLayout()
+        else:
+            cb_glay = self.clearLayout(cb_glay)
+
+        widget = self.getCtrlWidget(pv_name, '_L', config)
+        widget.clicked.connect(
+            lambda: self.controlBox(pv_name, cb_glay, config, True))
+        cb_glay.addWidget(widget, 1, 0, 1, 1)
+
+        return cb_glay
+
+    def controlBox(self, pv_name, lay, config, hasTitle):
+        ''' Display the control features - Open control Box'''
+
+        if lay != '':
+            lay = self.clearLayout(lay)
+        else:
+            lay = QGridLayout()
+        pos = [0, 0]
+        self.ledClicked = True
+        for control_name in CTRL_TYPE:
+            if hasTitle:
+                lb_title = QLabel(control_name)
+                lb_title.setAlignment(Qt.AlignCenter)
+                lay.addWidget(lb_title, pos[0], pos[1], 1, 1)
+            ctrl_widget = self.getCtrlWidget(
+                pv_name, CTRL_TYPE.get(control_name), config)
+            ctrl_widget.setMaximumWidth(100)
+            ctrl_widget.setMaximumHeight(60)
+            lay.addWidget(ctrl_widget, pos[0]+1, pos[1], 1, 1)
+            pos[1] += 1
+        return lay
+
+    def controlWidget(self, pv_name, lay, config):
+        '''Create the control widget with opening and closing functions'''
+        '''Default - closed'''
+
+        widget = QWidget()
+        vlay = QVBoxLayout()
+
+        if pv_name.find('WFS') != -1:
+            vlay.addWidget(
+                self.setPvLbl(pv_name))
+
+        control_layout = self.controlHiddenBox(pv_name, lay, config)
+        widget.setLayout(control_layout)
+        self.pv_obj.update(
+            {widget: {
+                "name": pv_name,
+                "layout": control_layout,
+                "config": config
+            }})
+        widget.installEventFilter(self)
+        widget.setObjectName('Container')
+        widget.setStyleSheet('''
+            #Container::hover{
+                background-color: #ff8b98;
+                border-radius: 5px;
+            }
+        ''')
+        vlay.addWidget(widget)
+        return vlay
+
+    def gateValve(self, pv_name, config):
+        ''' Display the gate valves widget '''
+        device_name = self.getDeviceName(pv_name)
+        if pv_name.find('U') != -1:
```

Comment:
since you don't need to know where in the string the substring `'U'` is, you could do `'U' in pv_name` instead. It is faster this way and I think it is more pythonic too.